### PR TITLE
Add ArrayBufferObject malloc, free handler into Platform

### PR DESCRIPTION
**Pull request by ksh8281**
Signed-off-by: Seonghyun Kim <sh8281.kim@samsung.com>

### CODE REVIEW

**Review by clover2123 — APPROVED** [review]
LGTM

File: src/runtime/Platform.h [hunk 1]
```diff
@@ -32,6 +32,10 @@ class Job;
 class Platform : public gc {
 public:
     virtual ~Platform() {}
+    // ArrayBuffer
+    virtual void* onArrayBufferObjectDataBufferMalloc(size_t sizeInByte) = 0;
+    virtual void onArrayBufferObjectDataBufferFree(void* buffer, size_t sizeInByte) = 0;
```
Comment (bbrto21) [inline src/runtime/Platform.h:37]: IMO
```suggestion
    virtual void* onMallocArrayBufferObjectDataBuffer(size_t sizeInByte) = 0;
    virtual void onFreeArrayBufferObjectDataBuffer(void* buffer, size_t sizeInByte) = 0;
```

File: src/api/EscargotPublic.h [hunk 2]
```diff
@@ -1777,6 +1777,18 @@ class ESCARGOT_EXPORT PlatformRef {
 public:
     virtual ~PlatformRef() {}
 
+    // ArrayBuffer
+    // client must returns zero-filled memory
+    virtual void* onArrayBufferObjectDataBufferMalloc(size_t sizeInByte)
+    {
+        return calloc(sizeInByte, 1);
+    }
```
Comment (bbrto21) [inline src/api/EscargotPublic.h:1785]: ```Malloc``` is used in the name of this method, but ```calloc``` is called actually.

File: src/runtime/Platform.h (hunk 1 above)
Comment (ksh8281) [reply src/runtime/Platform.h:37]: That's Good idea

File: src/api/EscargotPublic.h (hunk 2 above)
Comment (ksh8281) [reply src/api/EscargotPublic.h:1785]: IMO malloc is better.
calloc means contiguous alloc. but I want just zero-filled memory

File: src/runtime/Platform.h (hunk 1 above)
Comment (ksh8281) [reply src/runtime/Platform.h]: I updated pr

**Review by bbrto21 — APPROVED** [review]
Good!

**Pull request merged**